src/lashup_gm.erl

```diff
@@ -113,28 +112,30 @@ handle_info(_Down = {'DOWN', MonitorRef, _Type, _Object, _Info}, State) when is_
 handle_info({lashup_hyparview_events, #{type := current_views, ref := EventRef, active_view := ActiveView}},
   State0 = #state{hyparview_event_ref = EventRef}) ->
   State1 = handle_current_views(ActiveView, State0),
-  {noreply, State1};
+  {noreply, start_gc_timer(State1)};
 handle_info({nodedown, Node}, State) ->
   State1 = handle_nodedown(Node, State),
-  {noreply, State1};
+  {noreply, start_gc_timer(State1)};
 handle_info(trim_nodes, State) ->
   trim_nodes(State),
-  {noreply, State};
-handle_info(Info, State) ->
-  lager:debug("Unknown info: ~p", [Info]),
+  {noreply, start_gc_timer(State)};
+handle_info({timeout, GCRef, gc}, State = #state{gc_ref = GCRef}) ->
+  {noreply, State#state{gc_ref = undefined}, hibernate};
+handle_info(_Info, State) ->
   {noreply, State}.
 
-terminate(Reason, State) ->
-  lager:debug("Lashup_GM terminated, because: ~p, in state: ~p", [Reason, State]),
-  ok.
-
-code_change(_OldVsn, State, _Extra) ->
-  {ok, State}.
-
 %%%===================================================================
 %%% Internal functions
 %%%===================================================================
 
+-spec(start_gc_timer(state()) -> state()).
+start_gc_timer(#state{gc_ref = undefined} = State) ->
```

Comment:
how about placing this function in some `util` module and calling it from there? I see that it has been repeated across all the files. If some change has to be done in the future it has to be again repeated across all the files.

Reply:
`#state{}` are different in each of those files. Let me try to convert it to macro or store `gc_ref` in process dictionary (or in ets), but it could be even uglier.

Reply:
@GoelDeepak I've added a function that stores its state in process dictionary and returns `hibernate` one time every n seconds, otherwise it returns `infinity` (i.e. infinity timeout). Please let me know what you think about this approach.

Reply:
❤️ . This is much cleaner. Thank you!